src/main/scala/viper/gobra/ast/frontend/Ast.scala

```diff
@@ -194,6 +195,31 @@ case class PTypeDef(right: PType, left: PIdnDef) extends PTypeDecl
 
 case class PTypeAlias(right: PType, left: PIdnDef) extends PTypeDecl
 
+case class PConstructDecl(
+                          typ: PType, 
+                          args: Vector[PParameter], 
+                          posts: Vector[PExpression], 
+                          body: Option[(PBodyParameterInfo, PBlock)],
+                          isShared: Boolean
+                         ) extends PGhostMember with PScope
+
+case class PDerefDecl(
+                      typ: PType, 
+                      args: Vector[PParameter],
+                      result: PResult, 
+                      pres: Vector[PExpression], 
```

Comment:
Since we can put preconditions and postconditions in all the constructors (PConstructDecl, PDerefDecl, PAssignDecl), I defined a new PConstructSpec class, which takes: 
```
case class PConstructSpec(
                      pres: Vector[PExpression],
                      preserves: Vector[PExpression],
                      posts: Vector[PExpression],
                      isShared: Boolean
                      ) extends PSpecification
```

Reply:
ok